command/commands/token.py

```diff
@@ -0,0 +1,77 @@
+"""Command to obtain signed authentication token."""
+import jwt
+import logging
+
+from datetime import datetime
+from model.permissions import Permissions
+
+
+class TokenCommand:
+    """Token command model class."""
+
+    command_name = "token"
+    desc = "Generate a signed token for use with the HTTP API"
+    permission_error = "You do not have the sufficient " \
+                       "permission level for this command!"
+    lookup_error = "Requesting user not found!"
+
+    def __init__(self, db_facade, config):
+        """
+        Initialize TokenCommand.
+
+        :param db_facade: Database connection
+        """
+        logging.info("Initializing TokenCommand instance")
+        self.facade = db_facade
+        self.expiry = config.get_expiry()
+        self.signing_key = config.get_signing_key()
+
+    def get_name(self):
+        """Return the command's name."""
+        return self.command_name
+
+    def get_desc(self):
+        """Return a description of the command."""
+        return self.desc
+
+    def handle(self, _command, user_id):
+        """Handle request for token."""
+        logging.debug("Handling token command")
```

Comment:
one thing I've discovered is that logs like this are *super* unhelpful, because you end up getting a mumbo jumbo of logs with no way of associating series of events. For example, 2 users calling the same endpoint might generate logs like this:

```
DEBUG handling
DEBUG handling
INFO handling with option
INFO handling without option
INFO something
WARNING oh no
INFO success
INFO success
```

gets weird trying to tell who's who - it might be helpful to have some sort of request ID that propagates down to functions and can be included in logs as a field

Reply:
Honestly `user_id` would probably work for that - that's passed directly from Slack to us, so it's as good as a request ID (and I highly doubt somebody would try spamming commands to such a point we can't tell their requests apart).

Reply:
that is, until we have users programmatically accessing APIs with tokens and whatnot

Reply:
But this is _strictly_ for the Slack command - somebody accessing programmatically won't ever arrive here.

Reply:
oh, i was under the assumption that whatever API you guys build my hook into the command framework to avoid rewriting functionality, like:

```sh
rocket2.ubclaunchpad.com/user/edit?member=rwblickhan
# /rocket user edit --member=rwblickhan
```

Reply:
Probably something like that - though I was specifically referring to `/rocket token`, which should reaaaally only be accessible via Slack 😛

Reply:
tricky though - ideally an API would return proper JSON structures (we return plain text in inertia and https://github.com/ubclaunchpad/inertia/pull/546 was... not fun to write...)

I might consider having these functions return objects with relevant information, like

```
{ token: "", expiry: "" }
```

then write a thin layer above the commands that handles things in one of two ways:

* throw the object straight to the caller if via API
* format object into a pretty object if via Slack

Reply:
> specifically referring to `/rocket token`, which should reaaaally only be accessible via Slack

oh yeah that's fair, I guess this is going on a bit more of a tangent about things in general 😋 just throwing some thoughts out here

Reply:
> then write a thin layer above the commands that handles things in one of two ways:
> 
>     * throw the object straight to the caller if via API
> 
>     * format object into a pretty object if via Slack

I was thinking something along these lines, yeah. That would require a bit of refactoring, but hopefully not _too_ much.